editor/plugins/tile_map_editor_plugin.cpp

```diff
@@ -565,20 +565,31 @@ void TileMapEditor::_draw_cell(int p_cell, const Point2i &p_point, bool p_flip_h
 		}
 
 	} else if (node->get_tile_origin() == TileMap::TILE_ORIGIN_CENTER) {
-		rect.position += node->get_cell_size() / 2;
-		Vector2 s = r.size;
+		Size2 cell_size = node->get_cell_size();
 
-		Vector2 center = (s / 2) - tile_ofs;
+		rect.position += tile_ofs;
 
-		if (p_flip_h)
-			rect.position.x -= s.x - center.x;
-		else
-			rect.position.x -= center.x;
+		if (p_transpose) {
```

Comment:
Aren't the two branches doing the exact same thing?

Reply:
One is subtracting, the other is adding

Reply:
How embarassing...
I'll push an updated PR once I've run the tests, just to make sure I didn't mess anything up with the change.